corehq/apps/case_search/models.py

```diff
@@ -328,6 +330,17 @@ def case_search_synchronous_web_apps_for_domain(domain):
     return False
 
 
+@quickcache(['domain'], timeout=24 * 60 * 60, memoize_timeout=60)
+def case_search_sync_cases_on_form_entry_enabled_for_domain(domain):
+    if not case_search_enabled_for_domain(domain):
+        return False
+
+    config = CaseSearchConfig.objects.get_or_none(pk=domain)
+    if config:
+        return config.sync_cases_on_form_entry
+    return False
```

Comment:
I recognize that this same pattern is used above in the method `case_search_synchronous_web_apps_for_domain`, and that abstracting the logic to determine if case search is enabled is maybe more readable and flexible, but I also don't love that in the happy path of this method (assuming cleared cache), we request the same object from the DB twice (once to check if case search is enabled, and then once to check if `sync_cases_on_form_entry` is enabled)

I think due to the precedent that is set, this is out of scope of this PR, but I do wonder if something like this is cleaner.
```suggestion
def case_search_sync_cases_on_form_entry_enabled_for_domain(domain):
    config = CaseSearchConfig.objects.get_or_none(pk=domain, enabled=True)
    return config.sync_cases_on_form_entry if config else False
```

On a separate note, is caching queries for `CaseSearchConfig` worth it? I genuinely don't know what our expectations are around when to cache methods and when it's fine not to, but I'd imagine that a request to retrieve a `CaseSearchConfig` isn't terribly expensive given there is only one per domain, and that the configs are indexed by domain/domain is the primary key. Curious to hear what others think though!